osu.Game/Screens/Play/Player.cs

```diff
@@ -227,6 +222,11 @@ private void load(AudioManager audio, APIAccess api, OsuConfigManager config)
             hudOverlay.HoldToQuit.Action = Exit;
             hudOverlay.KeyCounter.Visible.BindTo(RulesetContainer.HasReplayLoaded);
 
+            RulesetContainer.IsPaused.BindTo(pauseContainer.IsPaused);
+
+            // schedule to ensure we count any key presses from the current frame (which may affect gameplay).
+            RulesetContainer.IsPaused.ValueChanged += paused => Schedule(() => hudOverlay.KeyCounter.IsCounting = !paused);
```

Comment:
Hmm, the thing I said is not fixed.
Let me say again. ESC down -> ESC up -> Z down -> ESC down -> you can see key counter is highlighted but the number is not increased.
Using `Schedule` on the another `ValueChanged`, not here, fixes this though I'm not sure if it is the best way.

Reply:
yeah, there has to be a better way to do this. using schedule for inter-class dependencies is a bit ugly. i'll reassess.

Reply:
As it turns out, it seems this was not even needed. Could you please re-test? It feels correct at my end.

Reply:
Seems working fine.